### PR TITLE
Add configuration of filestream input

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR adds the configuration of `filestream` to the config file to Filebeat.

## Why is it important?

Expose `filestream`.

## Checklist

~~- [ ] My code follows the style guidelines of this project~~
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
- [x] I have made corresponding change to the default configuration files
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

File: filebeat/_meta/config/filebeat.inputs.reference.yml.tmpl [hunk 1]
```diff
@@ -231,6 +232,145 @@ filebeat.inputs:
...
+  # for harvesting. Specify 1s to scan the directory as frequently as possible
+  # without causing Filebeat to scan too frequently. Default: 10s.
+  #prospector.scanner.check_interval: 10s
+
+  # Exclude files. A list of regular expressions to match. Filebeat drops the files that
+  # are matching any regular expression from the list. By default, no files are dropped.
+  #prospector.scanner.exclude_files: ['.gz$']
+
+  # Expand "**" patterns into regular glob patterns.
+  #procpector.scanner.recursive_glob: true
```
Comment (andresrc) [inline filebeat/_meta/config/filebeat.inputs.reference.yml.tmpl:280]: Is this a typo? (`procpector`)

File: filebeat/filebeat.reference.yml [hunk 2]
```diff
@@ -618,6 +619,145 @@ filebeat.inputs:
...
+  # for harvesting. Specify 1s to scan the directory as frequently as possible
+  # without causing Filebeat to scan too frequently. Default: 10s.
+  #prospector.scanner.check_interval: 10s
+
+  # Exclude files. A list of regular expressions to match. Filebeat drops the files that
+  # are matching any regular expression from the list. By default, no files are dropped.
+  #prospector.scanner.exclude_files: ['.gz$']
+
+  # Expand "**" patterns into regular glob patterns.
+  #procpector.scanner.recursive_glob: true
```
Comment (andresrc) [inline filebeat/filebeat.reference.yml:667]: Ditto

File: filebeat/filebeat.yml [hunk 3]
```diff
@@ -62,6 +62,35 @@ filebeat.inputs:
   # Note: After is the equivalent to previous and before is the equivalent to to next in Logstash
   #multiline.match: after
 
+# filestream is an experimental input. It is going to replace log input in the future.
+- type: filestream
```
Comment (michalpristas) [inline filebeat/filebeat.yml:66]: agent probably does not support this input do we need to include it?

**Review by michalpristas — APPROVED** [review]
change looks ok

File: filebeat/filebeat.yml (hunk 3 above)
Comment (kvch) [reply filebeat/filebeat.yml:66]: We should not include it at the moment. The input is experimental, so there might be some changes.

**Pull request merged**